File: bintree_py/test/testbst02.py
from .testbase import print_tree
from binarytree import BinSearchTree


def do_task():
    bst = BinSearchTree()

    for value in [12, 39, 20, 7, 26, 45, 19, 8]:
        bst.insert(value)

    print('\n print tree')
    print_tree(bst)

    print('\n height:', bst.height())

    print('\n')

    for value in [12, 39, 20, 7, 26, 45, 19, 8]:
        bst.remove(value)

    print('\n print tree')
    print_tree(bst)

    print('\n height:', bst.height())

    print()

File: bintree_java/src/test/TestBst02.java
package test;


import binarytree.BinSearchTree;



public class TestBst02 extends TestBase {

    @Override
    public void doTask() {
        var bst = new BinSearchTree<Integer, BinNodeImp<Integer> >();

        for ( var value : new int[] {12, 39, 20, 7, 26, 45, 19, 8} )
            bst.insert(value);

        printTree(bst);
        System.out.println("height: " + bst.height());


        for ( var value : new int[] {12, 39, 20, 7, 26, 45, 19, 8} )
            bst.remove(value);

        printTree(bst);
        System.out.println("height: " + bst.height());


        System.out.println();
    }

}
